src/hotspot/share/jvmci/jvmciEnv.cpp

```diff
@@ -229,17 +235,20 @@ void JVMCIEnv::init_env_mode_runtime(JavaThread* thread, JNIEnv* parent_env) {
 }
 
 JVMCIEnv::JVMCIEnv(JavaThread* thread, JVMCICompileState* compile_state, const char* file, int line):
-    _throw_to_caller(false), _file(file), _line(line), _compile_state(compile_state) {
-  init_env_mode_runtime(thread, NULL);
+    _throw_to_caller(false), _file(file), _line(line), _attach_threw_OOME(false), _compile_state(compile_state) {
+  init_env_mode_runtime(thread, NULL, false);
```

Comment:
why is it only non-fatal for this single path?  Some comments explaining the reason for it being fatal in some paths and not in others would help.

Reply:
I would say something like:
```
// In case of OOME, there's a good chance a subsequent attempt to attach might succeed.
// Other errors most likely indicate a non-recoverable error in the JVMCI runtime.
```
Note also that we will soon implement a change that fatal errors on libjvmci compiler threads will simply discard the thread.

Reply:
I've added the comment.
The compile broker compilation can be safely restarted. The affected method will be compiled later when the compiler isolate has enough memory. The other code paths don't have this property. One possible place where we can handle the isolate OOME in a non-fatal way is the `CompilerToVM#attachCurrentThread`. In this place, we can use JNI to throw OOME to the caller.